source/util/convert_to_iso_string.ts

```diff
@@ -0,0 +1,20 @@
+function isIsoDate(str: string) {
+  if (!/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z/.test(str)) return false;
```

Comment:
Should we also encode strings in a non-UTC timezone?

Reply:
we're encoding strings in a non-utc timezone, e.g. `2023-01-01T01:00:00+01:00`, and converting to UTC. see tests.

Reply:
The previous regex wasn't matching them - now it looks better.